### PR TITLE
Refactor genesis state

### PR DESCRIPTION
**Pull request by alexplevako**

### CODE REVIEW

File: libraries/app/application.cpp [hunk 1]
```diff
@@ -334,29 +339,28 @@ class application_impl : public graphene::net::node_delegate
                 if (_options->count("replay-blockchain"))
                 {
                     ilog("Replaying blockchain on user request.");
-                    _chain_db->reindex(_data_dir / "blockchain", _shared_dir, _shared_file_size);
+                    _chain_db->reindex(_data_dir / "blockchain", _shared_dir, genesis_state, _shared_file_size);
```
Comment (abolinbsu) [inline libraries/app/application.cpp]: Not logical insert genesis between 2 parameters describing single entity

Comment (alexplevako) [reply libraries/app/application.cpp]: agree, fixed.

File: libraries/wallet/include/scorum/wallet/wallet.hpp [hunk 2]
```diff
@@ -345,8 +347,8 @@ class wallet_api
      *  @param json_meta JSON Metadata associated with the new account
      *  @param broadcast true if you wish to broadcast the transaction
      */
-    annotated_signed_transaction create_account(
-        string creator, string new_account_name, string json_meta, bool broadcast);
+    annotated_signed_transaction
```
Comment (abolinbsu) [inline libraries/wallet/include/scorum/wallet/wallet.hpp]: General rule to this file: objects should be passed by const reference. Needs in refactoring

File: tests/plugin_tests/main.cpp [hunk 3]
```diff
@@ -25,17 +25,17 @@
...
 {
-    std::srand(time(NULL));
-    std::cout << "Random number generator seeded to " << time(NULL) << std::endl;
-    const char* genesis_timestamp_str = getenv("SCORUM_TESTING_GENESIS_TIMESTAMP");
-    if (genesis_timestamp_str != nullptr)
-    {
-        SCORUM_TESTING_GENESIS_TIMESTAMP = std::stoul(genesis_timestamp_str);
-    }
-    std::cout << "SCORUM_TESTING_GENESIS_TIMESTAMP is " << SCORUM_TESTING_GENESIS_TIMESTAMP << std::endl;
+    //    std::srand(time(NULL));
```
Comment (abolinbsu) [inline tests/plugin_tests/main.cpp]: Why comment rather than removal?

**Pull request merged**